pkgs/by-name/ju/just/package.nix

```diff
@@ -87,16 +68,41 @@ rustPlatform.buildRustPackage rec {
     "--skip=shebang::run_shebang" # test case very rarely fails with "Text file busy"
   ];
 
-  postInstall = ''
-    mkdir -p $doc/share/doc/$name
-    mv ./book/en/build/html $doc/share/doc/$name
-    installManPage man/just.1
-
-    installShellCompletion --cmd just \
-      --bash completions/just.bash \
-      --fish completions/just.fish \
-      --zsh completions/just.zsh
-  '';
+  postInstall =
+    let
+      just = "${stdenv.hostPlatform.emulator buildPackages} $out/bin/just";
+      doc = buildPackages.rustPlatform.buildRustPackage {
+        inherit
+          pname
+          version
+          src
+          cargoHash
+          ;
+        nativeBuildInputs = [ buildPackages.mdbook ];
```

Comment:
Given this is a build dependency for a build dependency, it should come from pkgsBuildBuild. (But also if you just use "mdbook" splicing should do the right thing.)